pkg/background/gpol/generate_controller.go

```diff
@@ -146,9 +154,15 @@ func (c *CELGenerateController) ProcessUR(ur *kyvernov2.UpdateRequest) error {
 				continue
 			}
 			engineResponse.PolicyResponse.Rules = []engineapi.RuleResponse{*res.Result}
-			engineResponse = engineResponse.WithPolicy(engineapi.NewGeneratingPolicy(&res.Policy))
+			engineResponse = engineResponse.WithPolicy(engineapi.NewGeneratingPolicyFromLike(res.Policy))
 			if res.Result.Status() == engineapi.RuleStatusSkip {
 				c.eventGen.Add(event.NewPolicyExceptionEvents(engineResponse, *res.Result, event.GeneratePolicyController)...)
+				if c.needsReports(*trigger) &&
+					reportutils.IsPolicyReportable(engineapi.NewGeneratingPolicyFromLike(res.Policy).AsKyvernoPolicy()) {
+					if err := c.createReports(context.TODO(), *trigger, engineResponse); err != nil {
+						c.log.Error(err, "failed to create report")
+					}
+				}
```

Comment:
I think we should move this outside the loop instead of duplicating the same code snippet twice inside the inner loop.